sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportPlugin.kt

```diff
@@ -0,0 +1,154 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.MapPlugin
+import com.mapbox.maps.plugin.viewport.experimental.data.DefaultViewportTransitionOptions
+import com.mapbox.maps.plugin.viewport.experimental.data.FollowingViewportStateOptions
+import com.mapbox.maps.plugin.viewport.experimental.data.OverviewViewportStateOptions
+import com.mapbox.maps.plugin.viewport.experimental.state.FollowingViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.OverviewViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * The Viewport plugin allows to track objects on a map.
+ *
+ * It provides a structured approach to organizing camera management logic into states and transitions between them.
+ *
+ * at any given time, the viewport is either:
+ *  - idle (not updating the camera)
+ *  - in a state (camera is being managed by a ViewportState)
+ *  - transitioning (camera is being managed by a ViewportTransition)
+ */
+interface ViewportPlugin : MapPlugin {
+
+  /**
+   * Returns current [ViewportStatus].
+   *
+   * If current status is IDLE, returns null.
+   * @see addStatusObserver
+   */
+  val status: ViewportStatus?
+
+  /**
+   * Executes a transition to requested state.
+   *
+   * When started, goes to [ViewportTransition]
+   * and to the final [ViewportState] when ended.
+   *
+   * If transition is canceled, state goes to IDLE.
+   *
+   * @param targetState The target [ViewportState] to transition to.
+   * @param completionListener The listener to observe the completion state.
+   */
+  fun transitionTo(targetState: ViewportState, completionListener: CompletionListener? = null)
+
+  /**
+   * Immediately goes to IDLE state canceling all ongoing transitions.
+   */
+  fun transitionToIdle()
```

Comment:
I'm open to this. Maybe we could name it just `idle()` like a command.

Reply:
Renamed to `idle()`

Reply:
From discussion: decided to name it `idle()` (verb usage since this method is used for its side-effects)